### PR TITLE
Update Username Display and Related Messaging

### PR DESCRIPTION
**Pull request by Namaskar-1F64F**
Well, since my last fix was so well received, I decided to make it work more better hopefully. I noticed `/members` doesn't respect first/last names and I looked through the code and couldn't find any other place that seemed like it needed updating. There is one place that's using username to query the DB, but it seems like it's working fine with people who don't have a username set so far so 🦐

### CODE REVIEW

**Comment by swindesr** [comment]
Nice, but is the DB missing big chunks of user info or something?

File: pages/api/lib/_helpers.ts [hunk 1]
```diff
@@ -181,3 +181,15 @@ export const getSubmissionDates = () => {
...
+export const getDisplayName = (user: Member) => {
+  const { first_name, last_name, username } = user.about;
+  if (username) {
+    return username;
+  }
+  const userFullName = `${first_name || ''} ${last_name || ''}`.trim();
+  if (userFullName !== '') {
+    return userFullName;
+  }
+  return 'Anonymous';
```
Comment (Geczy) [inline pages/api/lib/_helpers.ts:194]: could you add this to the locales file?

File: pages/api/lib/_locale.en.ts [hunk 2]
```diff
@@ -70,9 +73,9 @@ export const NO_WINNING_GROUPS_MESSAGE = `You haven't won the update lottery in
...
-  usernames: Array<string>
-) => `Stoodbot member${usernames.length > 1 ? 's' : ''} in this chat:
+  users: Array<Member>
+) => `A member, at random, will be awarded the opportunity to submit an update for this group. The member${users.length > 1 ? 's' : ''} currently subscribed in this chat:
 
-${usernames.map((g) => `• ${g}`).join('\n')}
+${users.map((u) => `• ${getDisplayName(u)}`).join('\n')}
 
-Type /subscribe to join!`;
+To have a chance of being chosen, type /subscribe to join!`;
```
Comment (Geczy) [inline pages/api/lib/_locale.en.ts:81]: love the new messaging!

File: pages/api/lib/_lottery.ts [hunk 3]
```diff
@@ -84,7 +82,7 @@ export const setWinners = async () => {
         )
       );
 
-      promises.push(sendMsg(WINNER_GROUP_MESSAGE(userDisplayName), chatId));
+      promises.push(sendMsg(WINNER_GROUP_MESSAGE(user), chatId));
```
Comment (Geczy) [inline pages/api/lib/_lottery.ts:85]: much better 👏🏻

**Review by Geczy — CHANGES_REQUESTED** [review]
you're probably referring to this part, about username being queried

https://github.com/alwaysbegrowing/telegram-standup-bot/blob/master/pages/api/updates.ts#L109-L115

it's not terribly important, this is for the "more" button on a user, seen here:

https://github.com/alwaysbegrowing/telegram-standup-bot/blob/master/components/project.tsx#L63

a better idea would be to use `userId` perhaps. have you found people with anonymous usernames to have a broken "more" link?

**Comment by Geczy** [comment]
it looks like they are indeed broken, an empty username is passed to the tg link & the more button 

if you want to merge this sooner than later, i can fix this in a latter pr 

image

File: pages/api/lib/_helpers.ts (hunk 1 above)
Comment (Namaskar-1F64F) [reply pages/api/lib/_helpers.ts]: I moved it

File: pages/api/lib/_locale.en.ts [hunk 4]
```diff
@@ -1,6 +1,17 @@
...
+  if (username) {
+    return username;
+  }
+  const userFullName = `${first_name || ''} ${last_name || ''}`.trim();
+  if (userFullName !== '') {
+    return userFullName;
+  }
+  return 'Anonymous';
+}
+
```
Comment (Geczy) [inline pages/api/lib/_locale.en.ts:14]: can you move this function back to helpers and only put `Anonymous` to the locale.en.ts file? sorry for being unclear

**Review by Geczy — CHANGES_REQUESTED** [review]

File: pages/api/lib/_locale.en.ts (hunk 4 above)
Comment (Namaskar-1F64F) [reply pages/api/lib/_locale.en.ts:14]: If the intention is to have the locale file have something like `getDisplayName(user) || "Anonymous"` I think that's too messy and also might not actually happen ever. Do you mean something else? I'm still confused 😃

Comment (Geczy) [reply pages/api/lib/_locale.en.ts:14]: locale.en.ts is meant to contain strings in the english locale

the intention is `return ANONYMOUS` where ANONYMOUS is a variable in locale.en.ts containing string `Anonymous`

Comment (Namaskar-1F64F) [reply pages/api/lib/_locale.en.ts:14]: ah, gotcha 👍

**Comment by Namaskar-1F64F** [comment]
I never have seen or used the web interface 🙃

**Review by Geczy — APPROVED** [review]

**Pull request merged**